enhancements/network/dpu-ovnkube-operator.md

```diff
@@ -0,0 +1,200 @@
+---
+title: DPU OVNKube Operator
+authors:
+  - "@pliurh"
+reviewers:
+  - "@zshi-redhat"
+  - "@dcbw"
+approvers:
+  - 
+creation-date: 2021-09-07
+last-updated: 2021-09-07
+status: implementable
+---
+
+# DPU OVNKube Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+
+## Summary
+
+To facilitate the management of Nvidia BlueField-2 DPU, a two-cluster design is
+proposed. Under such design, a BlueFiled-2 card will be provisioned as a worker
+node of the ARM-based infra cluster, whereas the tenant cluster where the normal
+user applications run on, is composed of the X86 servers.
+
+The OVN-Kubernetes components are spread over the two clusters. On the tenant
+cluster side, the Cluster Network Operator is in charge of the management of the
+ovn-kube components. On the infra cluster side, we propose to create a new
+operator to be responsible for the life-cycle management of ovn-kube components
+and the necessary host network initialization on DPU cards.
+
+## Motivation
+
+Normally, the Cluster Network Operator watches the Network customer resource,
+then provision the ovn-kube components to the Openshift cluster.
+
+In the two-cluster design, some of the ovn-kube components have to run on BF2
+DPUs, but they are still part of the ovn-kube cluster running in the tenant
+cluster. To allow the ovn-kube components on different clusters to talk with
+each other, we need to ensure all the necessary configs and secrets are aligned
+with each other.
+
+### Goals
+
+- Design a new operator that
+  - manages the ovn-kube components that serve the tenant cluster.
+  - enable the switchdev mode on DPUs.
+  - add PF representor to the br-ex bridge, so that the x86 host can connect to
+  the cluster network of the tenant cluster.
+- Define the ovn-kube upgrade mechanism in such a deployment.
```

Comment:
For example, should this OVNKube operator fetch the image information from tenant cluster as well and use the image to spin up ovnkube-node in infra cluster?

Reply:
Can we assume the two clusters will always be in the same version, and upgrade together? Then we can let the operator use the images which are currently used in the infra cluster.

Reply:
I think we need to define the version compatibility between infra and tenant cluster, like what is supported and not supported.  For example, we may say that 4.x tenant is compatible with 4.x+1 infra or 4.x+n infra.

This would allow new feature be introduced without updating tenant cluster. for example, the update of rhcos image in infra cluster (openvswitch, driver etc) may bring in new features, Operator designed to run on infra cluster would also benefit from it.

Thinking loudly, if we were to move ovn-central to infra cluster, it would further decouple the infra and tenant in terms of cluster or ovnkube versions. which allow more features be added without updating tenant cluster.